src/gluonts/model/rotbaum/_preprocess.py

```diff
@@ -270,10 +287,21 @@ def __init__(
             num_samples=num_samples,
             **kwargs
         )
+
+        if one_hot_encode:
+            assert (isinstance(cardinality, List) and cardinality) or (
+                isinstance(cardinality, str) and cardinality != "ignore"
+            ), "You should set `one_hot_encode=True` if and only if cardinality is not empty or ignored"
+
+        assert cardinality in set(["ignore", "auto"]) or all(
```

Comment:
This looks like a check we want to do earlier.

I think you could just create an Enum for this. You can look at the pydantic documentation for an example.